src/diagnostics/core_diagnostics.jl

```diff
@@ -0,0 +1,475 @@
+    standard_name = "air_potential_temperature",
+    units = "K",
+    comments = "Potential temperature of air",
+    compute! = (out, state, cache, time) -> begin
+        thermo_params = CAP.thermodynamics_params(cache.params)
+        if isnothing(out)
+            return TD.dry_pottemp.(thermo_params, cache.ᶜts)
+        else
+            out .= TD.dry_pottemp.(thermo_params, cache.ᶜts)
+        end
+    end,
+)
+
+###
+# Air pressure (3d)
+###
+add_diagnostic_variable!(
+    short_name = "pfull",
+    long_name = "Pressure at Model Full-Levels",
+    units = "Pa",
+    comments = "Pressure of air",
+    compute! = (out, state, cache, time) -> begin
+        if isnothing(out)
+            return copy(cache.ᶜp)
+        else
+            out .= cache.ᶜp
+        end
+    end,
+)
+
+###
+# Vorticity (3d)
+###
+add_diagnostic_variable!(
+    short_name = "rv",
+    long_name = "Relative Vorticity",
+    standard_name = "relative_vorticity",
+    units = "s^-1",
+    comments = "Vertical component of relative vorticity",
+    compute! = (out, state, cache, time) -> begin
+        vort = @. Geometry.WVector(curlₕ(state.c.uₕ)).components.data.:1
+        # We need to ensure smoothness, so we call DSS
+        Spaces.weighted_dss!(vort)
+        if isnothing(out)
+            return copy(vort)
+        else
+            out .= vort
+        end
+    end,
+)
+
+
+###
+# Relative humidity (3d)
+###
+compute_hur!(out, state, cache, time) =
+    compute_hur!(out, state, cache, time, cache.atmos.moisture_model)
+compute_hur!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("hur", model)
+
+function compute_hur!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        return TD.relative_humidity.(thermo_params, cache.ᶜts)
+    else
+        out .= TD.relative_humidity.(thermo_params, cache.ᶜts)
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "hur",
+    long_name = "Relative Humidity",
+    standard_name = "relative_humidity",
+    units = "",
+    comments = "Total amount of water vapor in the air relative to the amount achievable by saturation at the current temperature",
+    compute! = compute_hur!,
+)
+
+###
+# Total specific humidity (3d)
+###
+compute_hus!(out, state, cache, time) =
+    compute_hus!(out, state, cache, time, cache.atmos.moisture_model)
+compute_hus!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("hus", model)
+
+function compute_hus!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        return TD.total_specific_humidity.(thermo_params, cache.ᶜts)
+    else
+        out .= TD.total_specific_humidity.(thermo_params, cache.ᶜts)
```

Comment:
```suggestion
        out .= state.c.ρq_tot ./ state.c.ρ
```
(This is not the same as the current diagnostics, but is better. If you want the diagnostics to be the same as the current one you can leave this unchanged - I can change it in a future PR.)